module/zfs/zfs_vnops.c

```diff
@@ -2607,6 +2607,12 @@ zfs_getattr_fast(struct inode *ip, struct kstat *sp)
 
 	mutex_enter(&zp->z_lock);
 
+	bzero(sp, sizeof (*sp));
+#if defined(HAVE_STAT_RESULT_MASK)
+	/* Must be set before the call to generic_fillattr() */
+	stat->result_mask = STATX_BASIC_STATS;
+#endif /* defined(HAVE_STAT_RESULT_MASK) */
+
```

Comment:
Yes, and this function and simple_getattr are both hooks for iop->getattr(), so we don't need to do it if simple_getattr() don't do it.